odl/discr/grid.py

```diff
@@ -390,15 +390,22 @@ def stride(self):
         >>> g.stride
         array([  1.,  nan])
         """
-        strd = []
-        for i in range(self.ndim):
-            if not self.is_uniform_byaxis[i]:
-                strd.append(float('nan'))
-            elif self.nondegen_byaxis[i]:
-                strd.append(self.extent[i] / (self.shape[i] - 1.0))
-            else:
-                strd.append(0.0)
-        return np.array(strd)
+        # Cache for efficiency instead of re-computing
+        try:
+            strd = self.__stride
+        except AttributeError:
```

Comment:
I don't like this pattern of trying to access some attribute of `self` and react on failure. I would clearly prefer to initialize the attribute to some nonsense ("sentinel") value like `None` and check for that instead of checking if it's there at all. The check `a is None` should incur no cost whatsoever and make this more robust.